src/main/java/de/filefighter/rest/domain/filesystem/rest/FileSystemRestService.java

```diff
@@ -52,8 +52,14 @@ public FileSystemRestService(FileSystemBusinessService fileSystemBusinessService
     }
 
     @Override
-    public ResponseEntity<FileSystemItem> findFileOrFolderByNameAndAccessToken(String name, String accessToken) {
-        return null;
+    public ResponseEntity<List<FileSystemItem>> downloadFileSystemEntity(List<Long> fsItemIds, String accessTokenValue) {
+        User authenticatedUser = authenticationService.bearerAuthenticationWithAccessToken(accessTokenValue);
+        Pair<List<FileSystemItem>, String> listStringPair = fileSystemBusinessService.downloadFileSystemEntity(fsItemIds, authenticatedUser);
+
+        HttpHeaders responseHeaders = new HttpHeaders();
+        responseHeaders.set("Access-Control-Expose-Headers", RestConfiguration.FS_DOWNLOAD_NAME_HEADER);
```

Comment:
This is actually not needed, because it is only relevant for the browser and not for the filehandler.